pyccel/ast/c_concepts.py

```diff
@@ -57,6 +58,68 @@ def obj(self):
         """
         return self._obj
 
+
+class PointerCast(TypedAstNode):
+    """
+    A class which represents the casting of one pointer to another.
+
+    A class which represents the casting of one pointer to another in C code.
+    This is useful for storing addresses in a void pointer or for storing.
+    This class is not necessary but avoids warnings being raised in the
+    translated code.
```

Comment:
I suggest to expand the explanation:
```suggestion
    Using this class is not strictly necessary to produce correct C code,
    but avoids compiler warnings about the implicit conversion of pointers.
```